plugins/compiler/src/main/kotlin/PluginFactoryGenerator.kt

```diff
@@ -58,21 +69,33 @@ class PluginFactoryGenerator(private val codeGenerator: CodeGenerator) {
             .initializer(descriptorInitializer)
             .build()
 
-        val companionObject = TypeSpec.companionObjectBuilder()
-            .addProperty(descriptorProperty)
-            .build()
-
         val descriptorDelegateProperty = PropertySpec.builder("descriptor", PluginDescriptor::class, KModifier.OVERRIDE)
             .delegate("Companion::descriptor")
             .build()
 
-        // Create the create function that initializes the plugin with the descriptor and the config object.
-        val createFunction = FunSpec.builder("create").apply {
+        // Create the create function that initializes the plugin with the descriptor and the config object, created
```

Comment:
Maybe put `crate` in backticks to make it less sound like a word-duplication? Similar below.